libs/core/langchain_core/runnables/context.py

```diff
@@ -0,0 +1,246 @@
+import asyncio
+import threading
+from collections import defaultdict
+from functools import partial
+from itertools import groupby
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    DefaultDict,
+    Dict,
+    List,
+    Mapping,
+    Optional,
+    Type,
+    TypeVar,
+    Union,
+)
+
+from langchain_core.runnables.base import (
+    Runnable,
+    RunnableSerializable,
+    coerce_to_runnable,
+)
+from langchain_core.runnables.config import RunnableConfig, patch_config
+from langchain_core.runnables.utils import ConfigurableFieldSpec, Input, Output
+
+T = TypeVar("T")
+Values = Dict[Union[asyncio.Event, threading.Event], Any]
+CONTEXT_CONFIG_PREFIX = "__context__/"
```

Comment:
Make the context prefix dynamic for different context local scope so no conflict would occurs.